test/Spec.hs

```diff
@@ -74,13 +76,20 @@ propParserBSPrinter = mkPropParserPrinter $ bsToTxt . BS.toLazyByteString . Outp
 mkPropParserPrinter :: (ExecutableDocument Name -> T.Text) -> (TestLimit -> Property)
 mkPropParserPrinter printer = \space ->
   withTests space $ property $ do
-    xs <- forAll genExecutableDocument
-    let rendered = printer xs
-    either onError (xs ===) $ Input.parseExecutableDoc rendered
+    someRandomDoc <- forAll genExecutableDocument
+    let rendered = printer <$> Input.parseExecutableDoc (printer someRandomDoc)
+    -- the reason why we render twice is to make sure that
+    -- when comparing we already "normalized" the block
+    -- string, so each reparsing of the blockstring will no
+    -- longer be different because the indentations are
+    -- already at the minimun possible.
+    let reRendered = printer <$> (Input.parseExecutableDoc =<< rendered)
+    case (rendered, reRendered) of
+        (Left  e, _      ) -> onError e
+        (_      , Left  e) -> onError e
+        (Right a, Right b) -> a === b
```

Comment:
Does this mean that this PR weakens the test suite?

Reply:
unfortunately. I’m really looking forward to finding a better way. Anything by comes to mind?

Reply:
In light of the recent discovery (#44) that the test suite of this library was inadequate, I think it's really important that we figure this one out.